src/components/UserSelector.tsx

```diff
@@ -1,40 +1,80 @@
-import React from 'react';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { getUsers } from '../utils/fetchClient';
+import { ProvideContext } from './StateContext';
+import { User } from '../types/User';
+import classNames from 'classnames';
 
 export const UserSelector: React.FC = () => {
+  const {
+    users,
+    setUsers,
+    selectedUser,
+    setSelectedUser,
+    setPosts,
+    setSelectedPost,
+  } = useContext(ProvideContext);
+  const [showUsers, setShowUsers] = useState(false);
+
+  const userRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, [setUsers]);
```

Comment:
including setUsers in the dependency array is redundant since it's stable and it's better for useEffect to run only once after initial render which is typically desired behavior for data fetching